cdap-app-fabric/src/main/java/io/cdap/cdap/config/PreferencesService.java

```diff
@@ -101,10 +115,9 @@ private void setConfig(EntityId entityId,
   /**
    * Validate the profile status is enabled and set the preferences
    */
-  private void setConfig(ProfileStore profileStore, PreferencesTable preferencesTable, EntityId entityId,
-                         Map<String, String> propertyMap)
-    throws NotFoundException, ProfileConflictException, BadRequestException, IOException {
-
+  private void setConfig(ProfileStore profileStore, PreferencesTable preferencesTable,
+                         @Name("entityId") EntityId entityId, Map<String, String> propertyMap) throws Exception {
+    authorizationEnforcer.enforce(entityId, authenticationContext.getPrincipal(), Action.EXECUTE);
```

Comment:
here entity id can be InstanceId(""), a hardcoded value for instance level preferences, will user have permission on this?

Reply:
Also, should he action be ADMIN or EXECUTE? what is the scenario we use ADMIN and when we use EXECUTE

Reply:
Fixed.